sys/fuchsia/clock.txt

```diff
@@ -4,13 +4,17 @@
 include <zircon/syscalls.h>
 include <zircon/syscalls/clock.h>
 
+resource zx_clock[zx_handle]
 resource zx_time[int64]: 0, ZX_TIME_INFINITE
 
+# ZX_CLOCK_ARGS_VERSIONV1 0x400000000000000
+# ZX_CLOCK_ARGS_VERSIONV2 0x800000000000000
+
 zx_clock_get_monotonic() (ignore_return)
-zx_clock_create(option flags[clock_create_opt], args ptr[in, zx_clock_create_args_v1], out ptr[out, zx_handle])
-zx_clock_read(handle zx_handle, now ptr[out, intptr]) (ignore_return)
-zx_clock_get_details(handle zx_handle, option const[0x400000000000000], details ptr[out, zx_clock_details_v1])
-zx_clock_update(handle zx_handle, option flags[clock_update_opt], args ptr[in, zx_clock_update_args_v2])
+zx_clock_create(option flags[clock_create_opt], args ptr[in, zx_clock_create_args_v1], out ptr[out, zx_clock])
+zx_clock_read(handle zx_clock, now ptr[out, intptr])
+zx_clock_get_details(handle zx_clock, option const[0x400000000000000], details ptr[out, zx_clock_details_v1])
+zx_clock_update(handle zx_clock, option flags[clock_update_opt], args ptr[in, zx_clock_update_args_v2])
 
 clock_create_opt = ZX_CLOCK_OPT_MONOTONIC, ZX_CLOCK_OPT_CONTINUOUS, ZX_CLOCK_OPT_AUTO_START
 clock_update_opt = 0x800000000000000, ZX_CLOCK_UPDATE_OPTION_SYNTHETIC_VALUE_VALID, ZX_CLOCK_UPDATE_OPTION_REFERENCE_VALUE_VALID, ZX_CLOCK_UPDATE_OPTION_RATE_ADJUST_VALID, ZX_CLOCK_UPDATE_OPTION_ERROR_BOUND_VALID
```

Comment:
It looks like `option` is only valid for `zx_clock_update` if the 0x800... (`ZX_CLOCK_ARGS_VERSION(2)`) mask is combined with one of the other options -- the others aren't valid alone, if I'm understanding correctly. I don't know the answer to this question (maybe syz-mutate will answer it?): does the input generator `or` these options together, or do you need to specify that somehow?

Reply:
I'd assume the fuzzer eventually would get that right after a few iterations w/ `flags`.
I don't see anything like `or` in the [docs](https://github.com/google/syzkaller/blob/master/docs/syscall_descriptions_syntax.md).
We'll know more once we have source coverage.